src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataSymbolMapExtensions.cs

```diff
@@ -72,6 +72,73 @@ internal static class TaintedDataSymbolMapExtensions
             return taintedTargets != null;
         }
 
+        /// <summary>
+        /// Faster IsSourceMethod(), before using PointsToAnalysis or ValueContentAnalysis.
+        /// </summary>
+        /// <param name="sourceSymbolMap">SourceInfos.</param>
+        /// <param name="method">Invoked ethod to be evaluated.</param>
+        /// <param name="arguments">Arguments of the method.</param>
+        /// <param name="isSourceMethod">Indicates that the invoked method is definitely a tainted data source.</param>
+        /// <param name="requiresPointsTo">Indicates that the invoked method requires PointsToAnalysis for further
+        /// evaluation.</param>
+        /// <param name="requiresValueContent">Indicates that the invoked method requires ValueContentAnalysis for further
+        /// evaluation.</param>
+        /// <returns>True if the invoked method is potentially a tainted data source.</returns>
+        public static bool IsSourceMethodFast(
+            this TaintedDataSymbolMap<SourceInfo> sourceSymbolMap,
+            IMethodSymbol method,
+            ImmutableArray<IArgumentOperation> arguments,
+            out bool isSourceMethod,
+            out bool requiresPointsTo,
+            out bool requiresValueContent)
+        {
+            isSourceMethod = false;
+            requiresPointsTo = false;
+            requiresValueContent = false;
+
+            foreach (SourceInfo sourceInfo in sourceSymbolMap.GetInfosForType(method.ContainingType))
```

Comment:
I would let @LLLXXXCCC review the contents of this method.

Reply:
I'll merge optimistically.  @LLLXXXCCC please yell if you see anything wrong.